src/main/java/rodain/inspection/SipContentDirectory.java

```diff
@@ -0,0 +1,145 @@
+
+        this.addEventHandler(TreeItem.branchExpandedEvent(), new EventHandler<TreeModificationEvent<Object>>() {
+            @Override
+            public void handle(TreeModificationEvent<Object> e) {
+                SipContentDirectory source = (SipContentDirectory) e.getSource();
+                if (source.isExpanded()) {
+                    ImageView iv = (ImageView) source.getGraphic();
+                    iv.setImage(folderExpandImage);
+                }
+            }
+        });
+
+        this.addEventHandler(TreeItem.branchCollapsedEvent(), new EventHandler<TreeModificationEvent<Object>>() {
+            @Override
+            public void handle(TreeModificationEvent<Object> e) {
+                SipContentDirectory source = (SipContentDirectory) e.getSource();
+                if (!source.isExpanded()) {
+                    ImageView iv = (ImageView) source.getGraphic();
+                    iv.setImage(folderCollapseImage);
+                }
+            }
+        });
+    }
+
+    public TreeNode getTreeNode(){
+        return treeNode;
+    }
+
+    public void skip(){
+        SipContentDirectory parent = (SipContentDirectory)getParent();
+        TreeNode parentTreeNode = parent.getTreeNode();
+        parentTreeNode.remove(treeNode.getPath()); //remove this treeNode from the parent
+        parentTreeNode.addAll(treeNode.getAllFiles()); // add this treeNode's children to this parent
+    }
+
+    public void flatten(){
+        treeNode.flatten();
+        getChildren().clear();
+        for(String path: treeNode.getKeys()){
+            SipContentFile file = new SipContentFile(Paths.get(path));
+            getChildren().add(file);
+        }
+        sortChildren();
+    }
+
+    public void sortChildren(){
+        ArrayList<TreeItem<Object>> aux = new ArrayList<>(getChildren());
+        Collections.sort(aux, comparator);
+        getChildren().setAll(aux);
+
+        for(TreeItem ti: getChildren()){
+            if(ti instanceof SipContentDirectory)
+                ((SipContentDirectory)ti).sortChildren();
+        }
+    }
+
+    private static Comparator createComparator(){
+        return new Comparator<TreeItem>() {
+            @Override
+            public int compare(TreeItem o1, TreeItem o2) {
+                if(o1.getClass() == o2.getClass()) { //sort items of the same class by value
+                    String s1 = (String) o1.getValue();
+                    String s2 = (String) o2.getValue();
+                    return s1.compareToIgnoreCase(s2);
+                }
+                //directories must appear first
+                if(o1 instanceof SipContentDirectory)
+                    return -1;
+                return 1;
+            }
+        };
+    }
+
+    @Override
+    public String getPath() {
+        return this.fullPath;
+    }
+
+    @Override
+    public SourceTreeItemState getState(){
+        return SourceTreeItemState.NORMAL;
+    }
+
+    @Override
+    public void ignore(){
```

Comment:
why are ignore() and unignore() two methods instead of ignore(Boolean)?
Furthermore, what is the method overriding?